tools/kprint/main.cpp

```diff
@@ -0,0 +1,71 @@
+#include "kllvm/parser/KOREScanner.h"
+#include "kllvm/parser/KOREParser.h"
+
+#include <iostream>
+#include <fstream>
+
+using namespace kllvm;
+using namespace kllvm::parser;
+
+void readMap(std::map<std::string, std::string> &result, std::ifstream &is) {
+  std::string line;
+  while(std::getline(is, line)) {
+    std::string att;
+    if (!std::getline(is, att)) {
+      break;
+    }
+    result[line] = att;
+  }
+}
+
+
+int main (int argc, char **argv) {
+  if (argc != 3) {
+    std::cerr << "usage: " << argv[0] << " <kompiled-dir> <pattern.kore>" << std::endl;
+  }
+
+  std::map<std::string, std::string> formats;
+  formats["kseq"] = "%1 ~> %2";
+  formats["dotk"] = ".";
+  formats["inj"] = "%1";
+  std::map<std::string, std::string> hooks;
+  std::set<std::string> assocs;
+  std::set<std::string> comms;
+  std::map<std::string, std::vector<std::string>> colors;
+
+  std::ifstream iformats(argv[1] + std::string("/format-att.txt"));
```

Comment:
Who is supposed to write these files? The frontend, the user, or something else? I think I could infer what most of them mean, but, although I could follow what happens with that data, I don't think I actually understand `color-att.txt`.

Reply:
It is generated by the frontend; see the accompanying PR.

Reply:
Ok, but why does it make sense to have multiple colors for the same thing, and then to use each of them once?

Reply:
well, you may want to color different parts of the same production differently. consider for example the following production:
```
syntax Stmt ::= "if" "(" Stmt ")" Block "else" Block
```

Presumably you may want to color the parentheses differently than the keywords.